src/GovUk.Education.ExploreEducationStatistics.Common/Services/BlobStorageService.cs

```diff
@@ -362,7 +370,12 @@ public async Task<bool> IsAppendSupported(IBlobContainer containerName, string p
 
             try
             {
-                await blob.CreateIfNotExistsAsync();
+                await _storageInstanceCreationUtil.CreateInstanceIfNotExistsAsync(
+                    _connectionString,
+                    AzureStorageType.Blob,
+                    containerName.Name,
+                    () => blob.CreateIfNotExistsAsync());
+
```

Comment:
This doesn't look right to me, because the intention of this method is to check if 'append' operations are supported, by attempting a ` CreateIfNotExistsAsync()` on the `AppendBlobClient`.

A few lines above, `var blobContainer = await GetBlobContainer(containerName);` will already add the container to the list of created blob storage instances in the creation util. The params of  `CreateInstanceIfNotExistsAsync`  are going to be exactly the same here, so the task `blob.CreateIfNotExistsAsync()` will never be executed and we will never test if we can create an append block blob or not.

Incidentally this method isn't used anymore (I think it was originally written to append log messages to a blob).

Reply:
That was easy - removed :)

Also removed:

* `IsAppendSupported`
* `SetMetadata`

And made `UploadText` private.

Reply:
I think without this, then we could have done away with needing the `Func<Task> createIfNotExists` param in IStorageInstanceCreationUtil, which was necessary to cover both types of "thing" being created, which might've changed the overall implementation of this helper quite a bit, but no matter!

Reply:
Thanks, good to se those others get removed 🙂